main/src/ca/uwaterloo/flix/language/phase/typer/RecordConstraintSolver2.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.typer
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.shared.Scope
+import ca.uwaterloo.flix.language.ast.{Kind, Name, RigidityEnv, Symbol, Type, TypeConstructor}
+import ca.uwaterloo.flix.language.phase.typer.ConstraintSolver2.{Progress, TypeConstraint}
+import ca.uwaterloo.flix.language.phase.unification.Substitution
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+object RecordConstraintSolver2 {
+
+  /**
+    * Unifies the two given record row types.
+    */
+  def solve(tpe1: Type, tpe2: Type)(implicit scope: Scope, tracker: Progress, renv: RigidityEnv, flix: Flix): (List[TypeConstraint], Substitution) = (tpe1, tpe2) match {
+
+    // ----------
+    // ρ ~ ρ => ∅
+    case (t1, t2) if t1 == t2 =>
+      tracker.markProgress()
+      (Nil, Substitution.empty)
+
+    //    α ∉ fv(ρ)
+    // ----------------
+    // α ~ ρ  =>  {α ↦ ρ}
+    case (Type.Var(sym, _), tpe) if !tpe.typeVars.exists(_.sym == sym) && renv.isFlexible(sym) =>
```

Comment:
Perhaps we need to bite the bullet and add `contains`?

Reply:
You mean Type.contains?

Reply:
Yes

Reply:
Thoughts?

Reply:
Actually, does tpe.typeVars.contains not work?

Reply:
It works, but I am against it on principle :)